src/software/ai/hl/stp/tactic/goalie/goalie_tactic.cpp

```diff
@@ -0,0 +1,64 @@
+#include "software/ai/hl/stp/tactic/goalie/goalie_tactic.h"
+
+#include "software/ai/evaluation/calc_best_shot.h"
+#include "software/ai/hl/stp/action/stop_action.h"  // TODO (#1888): remove this dependency
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/point.h"
+
+GoalieTactic::GoalieTactic(std::shared_ptr<const GoalieTacticConfig> goalie_tactic_config)
+    : Tactic(true,
+             {RobotCapability::Move, RobotCapability::Dribble, RobotCapability::Chip}),
+      fsm(),
+      goalie_tactic_config(goalie_tactic_config),
+      control_params{
+          GoalieFSM::ControlParams{.goalie_tactic_config = goalie_tactic_config,
+                                   .clear_ball_origin    = std::nullopt,
+                                   .clear_ball_direction = std::nullopt}}
+{
+}
+
+void GoalieTactic::updateWorldParams(const World &world) {}
+
+void GoalieTactic::updateControlParams(
+    std::shared_ptr<const GoalieTacticConfig> updated_goalie_tactic_config,
+    std::optional<Point> clear_ball_origin, std::optional<Angle> clear_ball_direction)
+{
+    // Update the control parameters stored by this Tactic
+    control_params.goalie_tactic_config = updated_goalie_tactic_config;
+    control_params.clear_ball_origin    = clear_ball_origin;
+    control_params.clear_ball_direction = clear_ball_direction;
+}
```

Comment:
Would these parameters change over the course of a play? Could these just be passed into the constructor?

Reply:
Yeah, these can probably just be passed in the constructor - also if I remove the optional clear origin and direction overrides, then there'll just be the goalie tactic config, anyways